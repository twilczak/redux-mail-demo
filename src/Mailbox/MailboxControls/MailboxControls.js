import React, { Component } from 'react';
import { NavLink, Link } from 'react-router-dom';

import './MailboxControls.css';

export class MailboxControls extends Component {
    render() {
        return (
            <div>
                <NavLink to="/inbox" className="mailbox-button" activeClassName="is-active">Inbox</NavLink>
                <NavLink to="/outbox" className="mailbox-button" activeClassName="is-active">Outbox</NavLink>
                <Link to={this.props.match.path + "/compose"} className="mailbox-button">Compose</Link>
            </div>
        );
    }
}