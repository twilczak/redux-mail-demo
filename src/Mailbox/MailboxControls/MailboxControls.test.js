import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom';
import { MailboxControls } from './MailboxControls';

describe( MailboxControls, () => {
    const component = mount(
        <MemoryRouter>
            <MailboxControls match={{path: '/inbox'}}/>
        </MemoryRouter>
    );

    it('renders and matches the snapshot', () => {
        const component = renderer.create(
            <MemoryRouter>
                <MailboxControls match={{path: '/inbox'}}/>
            </MemoryRouter>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders /inbox link', () => {
        const inboxLink = component.find('.mailbox-button').at(0);
        expect(inboxLink.text()).toContain('Inbox');
        expect(inboxLink.getDOMNode().href).toContain('/inbox');
    });

    it('renders /outbox link', () => {
        const outboxLink = component.find('.mailbox-button').at(4);
        expect(outboxLink.text()).toContain('Outbox');
        expect(outboxLink.getDOMNode().href).toContain('/outbox');
    });

    it('renders compose link', () => {
        const outboxLink = component.find('.mailbox-button').at(6);
        expect(outboxLink.text()).toContain('Compose');
        expect(outboxLink.getDOMNode().href).toContain('/inbox/compose');
    });
});