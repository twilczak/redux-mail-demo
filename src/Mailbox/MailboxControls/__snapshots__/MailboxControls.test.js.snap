// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class MailboxControls extends _react.Component {
    render() {
        return (
            _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 9 }, __self: this },
                _react2.default.createElement(_reactRouterDom.NavLink, { to: '/inbox', className: 'mailbox-button', activeClassName: 'is-active', __source: { fileName: _jsxFileName, lineNumber: 10 }, __self: this }, 'Inbox'),
                _react2.default.createElement(_reactRouterDom.NavLink, { to: '/outbox', className: 'mailbox-button', activeClassName: 'is-active', __source: { fileName: _jsxFileName, lineNumber: 11 }, __self: this }, 'Outbox'),
                _react2.default.createElement(_reactRouterDom.Link, { to: this.props.match.path + "/compose", className: 'mailbox-button', __source: { fileName: _jsxFileName, lineNumber: 12 }, __self: this }, 'Compose')));


    }} renders and matches the snapshot 1`] = `
<div>
  <a
    aria-current={false}
    className="mailbox-button"
    href="/inbox"
    onClick={[Function]}
    style={undefined}
  >
    Inbox
  </a>
  <a
    aria-current={false}
    className="mailbox-button"
    href="/outbox"
    onClick={[Function]}
    style={undefined}
  >
    Outbox
  </a>
  <a
    className="mailbox-button"
    href="/inbox/compose"
    onClick={[Function]}
  >
    Compose
  </a>
</div>
`;
